src/handlers/handleIngressAPI.ts

```diff
@@ -22,7 +30,34 @@ async function makeIngressRequest(receivedRequest: Request, env: IntegrationEnv)
   addProxyIntegrationHeaders(request.headers, receivedRequest.url, env)
 
   console.log(`sending ingress request to ${url.toString()}...`)
-  return fetch(request, { backend: 'fpjs' })
+
+  if (!isOpenClientResponseEnabled(env)) {
+    return fetch(request, { backend: 'fpjs' })
+  }
+
+  const secretStore = new SecretStore(FingerprintSecretStoreName)
+  const decryptionKey = await secretStore.get(FingerprintDecryptionKeyName).then((v) => v?.plaintext())
+  if (!decryptionKey) {
+    throw new Error('Decryption key not found in secret store')
+  }
+
+  try {
+    const response = await fetch(request, { backend: 'fpjs' })
+
+    // Parse the open response
+    const text = await response.text()
+    const event = await unsealData(JSON.parse(text).sealedResult, decryptionKey)
```

Comment:
I'd prefer to not repeat `JSON.parse(text).sealedResult` code, instead introduce a local variable

Reply:
Refactored by this commit: https://github.com/fingerprintjs/fingerprint-pro-fastly-proxy-integration-example/pull/5/commits/95287e510ec4265368a4a720eeb97be371794b34